pyquil/paulis.py

```diff
@@ -324,33 +325,44 @@ def from_list(cls, terms_list, coefficient=1.0):
         pterm.coefficient = complex(coefficient)
         return pterm
 
+    @classmethod
+    def from_str(cls, str_pauli_term):
+        """Construct a PauliTerm from the result of str(pauli_term)
+        """
+        coef_str, str_pauli_term = str_pauli_term.split('*')
+        try:
+            coef = int(coef_str)
+        except ValueError:
+            try:
+                coef = float(coef_str)
+            except ValueError:
+                coef = complex(coef_str)
+
+        op = sI() * coef
+        for ma in re.finditer(r'([XYZ])(\d+)', str_pauli_term):
+            op *= PauliTerm(ma.group(1), int(ma.group(2)))
```

Comment:
I think we could just use `cls` here instead of `PauliTerm`

Reply:
is that better?

Reply:
I'm not sure if it makes a big difference, but I've usually seen classmethods used with the input class (otherwise this could've also just been a staticmethod, for example).

Reply:
done